openshift-kube-apiserver/handlers/deprecatedapirequest/update_manager.go

```diff
@@ -0,0 +1,62 @@
+package deprecatedapirequest
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/openshift-kube-apiserver/handlers/deprecatedapirequest/v1helpers"
+)
+
+// newUpdateManager creates and starts a new updateManager
+func newUpdateManager(client v1helpers.DeprecatedAPIRequestClient, resourceName string) *updateManager {
+	u := &updateManager{
+		client:   client,
+		resource: resourceName,
+	}
+	go wait.Forever(func() {
+		u.process(context.Background())
+	}, 1*time.Second)
+
+	return u
+}
+
+// updateManager maintains a queue of updates to apply when asked to.
+type updateManager struct {
```

Comment:
this looks familiar. Did I see this in apiextensions server?

Reply:
I've created something similar in kas-o for the check-endpoints command.